### PR TITLE
Fixed knex #1577

### PR DESCRIPTION
**Pull request by statyan**
Also fix an issue with `dropPrimary` for MSSQL and fix test for primary key - now column test2 is set to notNullable. Without it primarykey test fails 'cause nullable column can't be set for primary key.

### CODE REVIEW

**Comment by statyan** [comment]
Oh, don't matter about 3 commits "Fix for installing from github". It do nothing to code. Have no idea why they are listed for this PR, I thought I've added them later.

File: test/unit/schema/mssql.js [hunk 1]
```diff
@@ -76,11 +76,11 @@ describe("MSSQL SchemaBuilder", function() {
 
   it('test drop primary', function() {
     tableSql = client.schemaBuilder().table('users', function() {
-      this.dropPrimary();
+      this.dropPrimary('testconstraintname');
     }).toSQL();
 
     equal(1, tableSql.length);
-    expect(tableSql[0].sql).to.equal('ALTER TABLE [users] DROP PRIMARY KEY');
+    expect(tableSql[0].sql).to.equal('ALTER TABLE [users] DROP CONSTRAINT [testconstraintname]');
```
Comment (rhys-vdw) [inline test/unit/schema/mssql.js]: Is `DROP PRIMARY` not supported by MSSQL?

**Comment by rhys-vdw** [comment]
@statyan Looks good, I don't know much about MSSQL, but I'll cross reference these changes with the docs in future. Would you be able to squash these commits into one please?

File: test/unit/schema/mssql.js (hunk 1 above)
Comment (statyan) [inline test/unit/schema/mssql.js]: Yep, DROP CONSTRAINT only. See this article, section "Using Transact-SQL"
https://msdn.microsoft.com/en-us/library/ms190621.aspx

**Comment by statyan** [comment]
Ok,I'll try do it!

**Comment by elhigu** [comment]
@rhys-vdw when you click "merge pull request" button there is option to do squash merge if one likes.

**Comment by rhys-vdw** [comment]
@elhigu Of course, cheers.

Thanks @statyan!

**Pull request merged**